types-test/generate.mjs

```diff
@@ -0,0 +1,26 @@
+import fs from 'fs';
+import path from 'path';
+
+const ROOT_DIR = import.meta.dirname;
+
+const LINE_BREAK = '\n';
+
+const generatedTest = fs
+  .readFileSync(path.join(ROOT_DIR, 'generated-file.ts'), { encoding: 'utf8' })
+  .split(LINE_BREAK) // remove imports from file
+  .slice(4)
```

Comment:
This is very fragile

I would just use templating, (have a special string in each file that I would replace with the tests

Reply:
I'm kind of confused, what do you mean by templating?
You want me to have a special string at the end of each import line for example:
```js
import '../expect.d.ts'; // #####
import axios from 'axios'; // #####
import expect from 'expect'; // #####
```

Then look if the line ends with `// #####` and remove it accordingly?
Or how else would you do it?

Reply:
No,

Each type test file will be:

```ts
import axios from 'axios';

async function run() {
	'replace-me';
}
```

and your code will look for the replace-me string and replace it with the test code

Reply:
Although this way the files must be committed, so we can add a test in the script for replacing the content that you must have the `replace-me` string to avoid committing this code

Reply:
But if we have a file with just `expect` statements we won't have any syntax highlighting and auto-completion, which is the reason for it being a typescript file.

Also the problem you mentioned above, this seems problematic, don't you think?

Reply:
I see, you are right, but we need to find a better way for the imports

Reply:
Is the main problem with the inclusion of imports or deletion of them like I did.
Because what we can do for deletion is look for lines that have import at the start of them.
If it's inclusion that might be more problematic to implement